python/pyarrow/_dataset.pyx

```diff
@@ -134,6 +135,31 @@ cdef class DefaultPartitionScheme(PartitionScheme):
 
 
 cdef class SchemaPartitionScheme(PartitionScheme):
+    """
+    A PartitionScheme based on a specified Schema.
```

Comment:
There is still discussion to remove those classes, but already wrote those docstrings last week, so including here any way.

Reply:
Either way, we can reuse these docstring because the schemes must be explained.